### PR TITLE
add information about server side rendering

### PR DESCRIPTION
**Pull request by Slashek**

### CODE REVIEW

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Check Authorization Policies
+
+The next step is authorization using [Authorization Policies](https://documentation.platform-os.com/tutorials/authorization-policy/authorization-policy). Those are processed in the order in which we were defined - from top to bottom. If the authorization fails, the server stops processing the form and handles authorization policy failure(usually redirect or rendering 403 error).
+
+## ResourceOwner authorization
+
+Once all authorization policies pass, the next check is authorization ResourceOwner property. If fails, the server returns 404 error.
+
+## Decorating user input with `default_payload` json
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]:
```suggestion
## Modifying user input with `default_payload` json
```

**Review by pavelloz — CHANGES_REQUESTED** [review]
please use relative urls in your links (so that they work on staging as well)

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+If validation pass, the changeset between existing state of the resource and user input is calculated, and accessible from now on via `changes` variable.
+
+## Persisting changes
+
+The new state of the resource is persisted in the database. In case of rare scenario of internal server error, the changes are rollbacked and server renders 500 error.
+
+## Processing Callbacks
+
+The next step is to process the callback synchronously.
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid:61]: It looks like this article could use an ordered list or at least numbers before each step

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+## Schedule processing AsyncCallbacks
+
+The next step is to schedule pocessing async callbacks in the background.
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]:
```suggestion
The next step is to schedule processing async callbacks in the background.
```

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+## Schedule sending notifications
+
+The next step is to schedule all notifications: email, sms, api calls, to be processed in the background
```
Comment (pavelloz) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]:
```suggestion
The next step is to schedule all notifications (email, sms, api calls) to be processed in the background.
```

**Comment by Slashek** [comment]
@pavelloz updated, and added one note and one tip

**Review by pavelloz — APPROVED** [review]

**Comment by pavelloz** [comment]
@diana-lakatos - merge if you decide its ready :)

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 5]
```diff
@@ -0,0 +1,78 @@
+---
+converter: markdown
+metadata:
+  title: Form Server Side Processing
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: Server-Side

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+{% endcapture %}
+{% include 'alert/note', content: content %}
+
+## 5. Modifying user input with `default_payload` json
+
+The next step is to extend/overwrite user's input with `default_payload`. This is the place when liquid is evaluated server side. The most useful variables here used to build default_payload are `context.params` and `form` variables. If evaluated liquid is not valid json, the server raises 500 error and automatically creates relevant [Log](/tutorials/qa/creating-marketplace-kit-logs).
+
+## 6. Validating user input
+
+The result of merging user's input with default_payload is then run against validation rules defined for each field. If at least one validation fails, the server renders response based on the format. For html the errors are added to `form_builder` variable inside the relevant field and server renders [Page](/tutorials/pages/pages) based on `page_id` parameter passed to the server. Additionally, the flash_alert is evaluated and set if provided in the configuration. It's then accessible using `context` variable. For JSON, a json object is returned with populated `errors` key. The status code is 200 in both scenarios.
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]:
```suggestion
The result of merging user's input with default_payload is then run against validation rules defined for each field. If at least one validation fails, the server renders response based on the format. For HTML the errors are added to `form_builder` variable inside the relevant field and server renders [Page](/tutorials/pages/pages) based on `page_id` parameter passed to the server. Additionally, the flash_alert is evaluated and set if provided in the configuration. It's then accessible using `context` variable. For JSON, a json object is returned with populated `errors` key. The status code is 200 in both scenarios.
```

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+## 11. Schedule sending notifications
+
+The next step is to schedule all notifications (email, sms, api calls) to be processed in the background.
+
+## 12. Render response
+
+If the format is set to html, the flash_notice property is evaluated and set, then redirect_to property is evaluated to determine where the user should be redirected. For JSON request, the server renders 200 status and serializes the form to json few basic properties (like `id`) to json.
+
+{% capture content %}
+It is possible to send request via AJAX, but receive HTML response from the server. The trick is to explicitly set the format to html by appending to the rest endpoint `.html`, for example `/api/users.html`. The server will render html if validation fails (so you could display errors) or will return a json including redirect url upon success.
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: URL

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 8]
```diff
@@ -0,0 +1,78 @@
+---
+converter: markdown
+metadata:
+  title: Form Server Side Processing
+  description: Form Server Side Processing in PlatformOS.
+slug: tutorials/forms/form-server-side-processing
+searchable: true
+---
+This topic explains form server side processing in PlatformOS when user submits form.
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: platformOS

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+The very first thing before every POST request is to verify CSRF token. If it is not provided, the user's session won't be validated and will be discarded.
+
+{% capture content %}
+  For AJAX calls make sure to provide headers `X-CSRF-Token: <value from context.authenticity_token>` and `X-Requested-With: 'XMLHttpRequest`. If you use {% raw %}`{% form %}`{% endraw %} tag, then authencitiy_token is included automatically as a hidden input and no action is needed.
+{% endcapture %}
+{% include 'alert/important', content: content %}
+
+
+## 2. Find FormConfiguration based on `form_configuration_name` or `form_configuration_name` parameter
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: form_configuration_id ?

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+The first step is to identify FormConfiguration to be used to process user's request. In order to do it, the server checks for parameter called `form_configuration_name` or `form_configuration_id`, which is automatically added to the page when form is rendered using {% raw %}`{% form %}`{% endraw %} tag as described in [FormTag](/tutorials/forms/form-rendering#form-tag). If the value is missing or invalid, 404 error is rendered, otherwise we move on.
+
+## 3. Check Authorization Policies
+
+The next step is authorization using [Authorization Policies](/tutorials/authorization-policy/authorization-policy). Those are processed in the order in which we were defined - from top to bottom. If the authorization fails, the server stops processing the form and handles authorization policy failure(usually redirect or rendering 403 error).
+
+## 4. ResourceOwner authorization
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: I think better Resource Owner

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+Once all authorization policies pass, the next check is authorization ResourceOwner property. It checks if the currently logged in user has access to the resource with given `id`. If fails, the server returns 404 error.
+
+{% capture content %}
+ResourceOwner is only relevant for update / delete. It does not matter for create, as there is no resource yet.
+{% endcapture %}
+{% include 'alert/note', content: content %}
+
+## 5. Modifying user input with `default_payload` json
+
+The next step is to extend/overwrite user's input with `default_payload`. This is the place when liquid is evaluated server side. The most useful variables here used to build default_payload are `context.params` and `form` variables. If evaluated liquid is not valid json, the server raises 500 error and automatically creates relevant [Log](/tutorials/qa/creating-marketplace-kit-logs).
```
Comment (lemingos) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: JSON

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid (hunk 7 above)
Comment (lemingos) [reply marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: and HTML

Comment (lemingos) [reply marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: and JSON :)

**Review by lemingos — CHANGES_REQUESTED** [review]

File: marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+The result of merging user's input with default_payload is then run against validation rules defined for each field. If at least one validation fails, the server renders response based on the format. For html the errors are added to `form_builder` variable inside the relevant field and server renders [Page](/tutorials/pages/pages) based on `page_id` parameter passed to the server. Additionally, the flash_alert is evaluated and set if provided in the configuration. It's then accessible using `context` variable. For JSON, a json object is returned with populated `errors` key. The status code is 200 in both scenarios.
+
+## 7. Calculating changes
+
+If validation pass, the changeset between existing state of the resource and user input is calculated, and accessible from now on via `changes` variable.
+
+## 8. Persisting changes
+
+The new state of the resource is persisted in the database. In case of rare scenario of internal server error, the changes are rollbacked and server renders 500 error.
```
Comment (ghost) [inline marketplace_builder/views/pages/tutorials/forms/form-server-side-processing.liquid]: rollbacked -> rolled back

**Comment by Slashek** [comment]
updated globally "PlatformOS" -> "platformOS"

**Review by lemingos — APPROVED** [review]

**Review by pavelloz — CHANGES_REQUESTED** [review]
From what i remember Diana wanted to keep capitalized `P` if its in the beginning of the sentence

**Comment by Slashek** [comment]
hmm, @diana-lakatos ^

**Comment by diana-lakatos** [comment]
I discussed this with Adam, and we're going with `platformOS` even if it's in the beginning of the sentence to keep the branding consistent.

**Comment by pavelloz** [comment]
Good stuff :)

**Review by pavelloz — APPROVED** [review]

**Pull request merged**